### ISSUE TITLE
FileTool breaks contentlet rendering if the file/image is not live

### ISSUE THREAD

**Issue by dsolistorres** [dotCMS/core#18014]
Reported via Support: [Zendesk-100539](https://dotcms.zendesk.com/agent/tickets/100539)
Reproducible in dotCMS 5.2.5

For a content type that contains a field of image or file type, the Velocity generated code uses `FileTool` viewtool (`getFile` method) to retrieve the image or file. If the image/file is not live, `FileTool` throws an exception and the contentlet rendering breaks.

### Steps to reproduce the behavior:
1. Go to 'Content Types' portlet a create a new content type
2. Add a new field from File type to the content type (Image field also works)
3. Add a second field from Text type to the content type
4. Go to 'Containers' portlet and create a new container that shows the Text field created in the previous step
5. Go to 'Templates' portlet and create a new template that uses the container created in the previous step
6. Go to 'Site Browser' and add a new page using the template created in the previous step.
7. Add a new contentlet to the page created in the previous step. Choose an existing image for the File field and fill the Text field with some text.
8. Unpublish the file chosen for the contentlet in the previous step.
9. If you logout from dotCMS backend and go to the live page created in the step 7, the contentlet doens't render the Text field, and an error message shows in the error log:
```
[19/02/20 12:01:44:206 CST] ERROR directive.ContentletDetail: Exception rendering contentDetail (/LIVE/72403fcf-0021-44cc-98ab-089b6ffa8d21_1.contentlet) at /LIVE/b59a624d-6d06-487f-a109-d86c523a5670/dotParser_1582133967356.container[line 1, column 1561]. Cause of error: Invocation of method 'getFile' in  class com.dotcms.rendering.velocity.viewtools.FileTool threw exception com.dotmarketing.business.DotStateException: Contentlet is null at /LIVE/72403fcf-0021-44cc-98ab-089b6ffa8d21_1.contentlet[line 1, column 897] @ url:GET//demo.dotcms.com/training/file-and-test-test-page-1 | lang:1 | ip:0:0:0:0:0:0:0:1 | Admin:false | start:02-19-2020 11:22:48 CST
```

### Expected behavior
Contentlet should be rendered even if the file pointed by the File/Image field is not live.

### Screenshots
Content type with File and Text fields:
![image](https://user-images.githubusercontent.com/2894221/74861510-19768000-5310-11ea-8915-60df6289d562.png)

Container for the content type:
![image](https://user-images.githubusercontent.com/2894221/74861593-3d39c600-5310-11ea-81d1-c3794839b70e.png)

Template for the container:
![image](https://user-images.githubusercontent.com/2894221/74861612-462a9780-5310-11ea-85a7-79abfbf97663.png)

Contentlet pointing to a file:
![image](https://user-images.githubusercontent.com/2894221/74861670-5d698500-5310-11ea-8077-0b9e445908ba.png)

### Additional context
This has been reported by customer via Support: Ticket [100539](https://dotcms.zendesk.com/agent/tickets/100539).

**Comment by x0rsw1tch** [comment]
I just ran into this with a client site. They forgot to publish the images associated with image fields in a contentlet, preventing regular content types to render the output in front-end or live preview mode.

The widget Content Types threw the exception, but still managed to render properly.

Here is an excerpt from the log

`21/04/21 17:36:34:949 EDT] ERROR directive.ContentletDetail: Exception rendering contentDetail (/LIVE/aef00f1b-84ea-4957-aad5-b66e8e8b0587_1.contentlet) at /LIVE/%%www#thousandeyes#com%application%containers%main-content%/1591819661189.container[line 1, column 1655]. Cause of error: Invocation of method 'getFile' in  class com.dotcms.rendering.velocity.viewtools.FileTool threw exception com.dotmarketing.business.DotStateException: Contentlet is null at /LIVE/aef00f1b-84ea-4957-aad5-b66e8e8b0587_1.contentlet[line 1, column 1591] @ url:GET//www.thousandeyes.com/solutions/saas-monitoring | lang:1 | ip:76.189.141.244 | Admin:false | start:04-21-2021 05:35:28 EDT`

**Comment by wezell** [comment]
FileTool also breaks when the selected image/file is a dotAsset.

```
[12/07/21 14:08:19:922 GMT] ERROR directive.ContentletDetail: Exception rendering contentDetail (/PREVIEW_MODE/5ff0667a-ed21-4738-89cf-f34a101fa650_1.contentlet) at /PREVIEW_MODE/408bf75c-56fb-42f3-b337-7c6a6ae69b89/1620758809276.container[line 1, column 1531]. Cause of error: Invocation of method 'getFile' in class com.dotcms.rendering.velocity.viewtools.FileTool threw exception com.dotmarketing.business.DotStateException: Contentlet : add3fe40-adc0-4755-a32d-25d8b8eaa18b is not a FileAsset at /PREVIEW_MODE/5ff0667a-ed21-4738-89cf-f34a101fa650_1.contentlet[line 1, column 1250] @ url:GET//default/api/v1/page/render/index | lang:1 | ip:18.224.182.142 | Admin:false | start:07-12-2021 02:04:04 GMT ?language_id=1

```

**Comment by dsilvam** [comment]
Passed Internal QA: page showing text field value when the File is unpublished.

**Comment by bryanboza** [comment]
Fixed, tested on release 21.07 // Postgres // FF
